client/src/pages/HomePage.tsx

```diff
@@ -14,32 +15,50 @@ type Props = {
 function HomePage({ socket, auth }: Props) {
   const navigate = useNavigate();
   const [inputRoomID, setInputRoomID] = useState<string>("");
-  const { getRooms, addNewRoom } = useContext(RoomsContext);
-
-  const createRoom = useCallback(() => {
-    socket.emit(RoomEvent.CREATE_ROOM, { roomId: uuidv4(), userId: auth });
-  }, [socket, auth]);
+  const { addNewRoom } = useContext(RoomsContext);
+  const [err, setErr] = useState<boolean>(false);
+  const createRoom = useCallback(async () => {
+    try {
+      const { data } = await serverAxios.post("watch-app/rooms/", {
```

Comment:
again, I want to see these requests to the endpoints being "injected" "passed down" to the components instead of being defined in the component